utils/cPlatformLayerInit.cpp

```diff
@@ -0,0 +1,38 @@
+#include "cPlatformLayerInit.h"
+
+#include "allegroh.h"
+#include "utils/cLog.h"
+
+#include <cstring>
+#include <stdexcept>
+
+static bool allegroInitialized = false;
+
+cPlatformLayerInit::cPlatformLayerInit(const std::string& path_to_config_file) {
+    cLogger *logger = cLogger::getInstance();
+    logger->logHeader("Allegro");
+
+    if (allegroInitialized) {
+        throw std::runtime_error("Allegro initialization was attempted twice.");
```

Comment:
this is something new (I believe there is no other code path throwing any kind of exceptions). What is the consensus about this? Afaik exceptions throwing/catching is a no-no in C++ land?

Reply:
It's a way to cancel construction of an object. The only way (apart from flat-out calling exit ;) )
And you can let the exception catch anywhere in the stack, or nowhere. Some programs have some handler in main or somewhere else on top. Others catch all and return an error code.
In this case, I just let it flow. Interestingly, the console output shows the exception. Exactly what I wanted and without any additional coding. :)